userguide/sound_classifier/export-coreml.md

```diff
@@ -0,0 +1,114 @@
+It should look something like this:
+![Xcode ML Model Screen Shot](xcode-mlmodel.png)
+
+Before you can use the model you must download
+`libAudioPreprocessing.dylib`. This is file is needed for the signal
+preprocessing stage, which is implemented as a Core ML custom model.
+The `libAudioPreprocessing.dylib` file can be found on the
+[Github releases page](https://github.com/apple/turicreate/releases).
+Once you've download that file, drag and drop it into your Xcode project.
+The dialog box should look like:
+![Xcode dylib Screen Shot](xcode-dylib.png)
+
+Now, do the following:
+* Under **Project > Build Phases**, add `libAudioPreprocessing.dylib` to **Copy Bundle Resources**
+* Under **Project > General**, add `libAudioPreprocessing.dylib` to **Embedded Binaries** and to **Linked Frameworks and Libraries**
+
+
+Now you are ready to start using your model. To create an instance of
+your model simply do the following:
+```swift
+let model = my_sound_classifier()
+```
+
+The input to this model is an `MLMultiArray` of length 15,600 this is
+975ms of data at a sample rate of 16K per second (.975 * 16,000 = 15,600).
+It is important that the correct sample rate of 16,000 be used.
+
+In order get predictions from your model, you will need to chuck your
+data into the correct size. Below is an example of doing that:
+```swift
+// Read wav file
+var wav_file:AVAudioFile!
+do {
+   wav_file = try AVAudioFile(forReading:audioFilename)
+} catch {
+   fatalError("Could not open wav file.")
+}
+
+print("wav file length: \(wav_file.length)")
+assert(wav_file.fileFormat.sampleRate==16000.0, "Sample rate is not right!")
+
+let buffer = AVAudioPCMBuffer(pcmFormat: wav_file.processingFormat, frameCapacity: UInt32(wav_file.length))
+do {
+   try wav_file.read(into:buffer!)
+} catch{
+   fatalError("Error reading buffer.")
+}
+guard let bufferData = try buffer?.floatChannelData else {
+   fatalError("Can not get a float handle to buffer")
+}
+
+// Chunk data and set to CoreML model
+let windowSize = 15600
+guard let audioData = try? MLMultiArray(shape:[windowSize as NSNumber], dataType:MLMultiArrayDataType.float32) else {
+   fatalError("Can not create MLMultiArray")
+}
+
+// Ignore any partial window at the end.
+var windowNumber = 0
+var results = [Dictionary<String, Double>]()
+while ((windowNumber+1) * windowSize) <= (wav_file.length - Int64(windowSize)) {
+   let offset = windowNumber * windowSize
+   for i in 0...windowSize {
+      audioData[i] = NSNumber.init(value: bufferData[0][offset + i])
+   }
+   let modelInput = modelInput(audio: audioData)
+
+   guard let modelOutput = try? model.prediction(input: modelInput) else {
+      fatalError("Error calling predict")
+   }
+   results.append(modelOutput.categoryProbability)
+   windowNumber += 1
+}
+```
+
+Once all window predictions are saved, to generate the overall
+prediction, calculate the mean accross all prediciton windows:
+```
```

Comment:
Add `swift` at the end of the three backticks here for proper syntax highlighting?

Reply:
Good catch. Done.